library/core/class.oauth2.php

```diff
@@ -529,7 +533,7 @@ public function entryEndpoint($sender, $code, $state) {
         } elseif (empty($response['access_token'])) {
             throw new Gdn_UserException('The OAuth server did not return an access token.', 400);
         } else {
-            $this->accessToken($response['access_token']);
+            $this->accessToken(val('access_token', $response));
```

Comment:
Why changing this one and not the one above?
Also we already have a check the access_token exists above. Calling val() is not necessary. No big deal though.